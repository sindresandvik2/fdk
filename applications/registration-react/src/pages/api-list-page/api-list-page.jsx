import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Link } from 'react-router-dom';

import localization from '../../utils/localization';
import ListItems from '../../components/list-items/list-items.component';
import getTranslateText from '../../utils/translateText';

export const APIListPage = props => {
  const {
    catalogItem,
    items,
    fetchCatalogIfNeeded,
    fetchApisIfNeeded,
    match
  } = props;

  const catalogId = _.get(match, ['params', 'catalogId']);
  if (catalogId) {
    fetchCatalogIfNeeded(catalogId);
    fetchApisIfNeeded(catalogId);
  }

  return (
    <div className="container">
      <div className="row mb-5">
        <div className="col-12">
          <h1>{localization.api.register.apiCatalog}</h1>
          <div className="fdk-reg-datasets-publisher mt-2 mb-4">
            {getTranslateText(_.get(catalogItem, ['publisher', 'prefLabel'])) ||
              _.get(catalogItem, ['publisher', 'name'])}
          </div>
        </div>
      </div>
      <div className="row mb-5">
        <div className="col-12">
          <Link className="ml-0" to={`/catalogs/${catalogId}/apis/import`}>
            <i className="fa fa-plus mr-2" />
            {localization.api.register.registerNew}
          </Link>
        </div>
      </div>
      <div className="row mb-2 mb-md-5">
        {catalogItem && (
          <div className="col-12 fdk-reg-datasets-list">
            <ListItems
              catalogId={catalogId}
              items={items}
              itemTitleField={['openApi', 'info', 'title']}
              prefixPath={`/catalogs/${catalogId}/apis`}
              defaultEmptyListText={localization.listItems.missingApiItems}
            />
          </div>
        )}
      </div>
    </div>
  );
};

APIListPage.defaultProps = {
  catalogItem: null,
  items: null,
  fetchCatalogIfNeeded: () => {},
  fetchApisIfNeeded: () => {},
  match: null
};

APIListPage.propTypes = {
  catalogItem: PropTypes.object,
  items: PropTypes.array,
  fetchCatalogIfNeeded: PropTypes.func,
  fetchApisIfNeeded: PropTypes.func,
  match: PropTypes.object
};
