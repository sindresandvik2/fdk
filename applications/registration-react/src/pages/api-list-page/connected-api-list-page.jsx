import { connect } from 'react-redux';
import _ from 'lodash';
import { fetchApisIfNeededAction } from '../../redux/modules/apis';
import { fetchCatalogIfNeeded } from '../../redux/modules/catalog';
import { APIListPage } from './api-list-page';

const mapStateToProps = ({ catalog, apis }, ownProps) => {
  const catalogId = _.get(ownProps, ['match', 'params', 'catalogId']);
  const catalogItem = _.get(catalog, ['items', catalogId]);
  const items = _.get(apis, [catalogId, 'items']);
  return {
    catalogItem,
    items
  };
};

const mapDispatchToProps = dispatch => ({
  fetchCatalogIfNeeded: catalogURL =>
    dispatch(fetchCatalogIfNeeded(catalogURL)),
  fetchApisIfNeeded: catalogId => dispatch(fetchApisIfNeededAction(catalogId))
});

export const ConnectedAPIListPage = connect(
  mapStateToProps,
  mapDispatchToProps
)(APIListPage);
