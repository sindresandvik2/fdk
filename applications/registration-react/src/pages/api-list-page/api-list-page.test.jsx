import React from 'react';
import { shallow } from 'enzyme';

import { APIListPage } from './api-list-page';

let defaultProps;
let wrapper;
let fetchCatalogIfNeeded;
let fetchApisIfNeeded;

beforeEach(() => {
  fetchCatalogIfNeeded = jest.fn();
  fetchApisIfNeeded = jest.fn();
  const catalogItem = {
    id: '910244132',
    uri: 'http://brreg.no/catalogs/910244132',
    title: {
      nb: 'Datakatalog for RAMSUND OG ROGNAN REVISJON'
    },
    description: {},
    publisher: {
      uri: 'http://data.brreg.no/enhetsregisteret/enhet/910244132',
      name: 'RAMSUND OG ROGNAN REVISJON'
    }
  };
  const match = {
    path: '/catalogs/:catalogId/apiSpecs',
    url: '/catalogs/910244132/apiSpecs',
    isExact: true,
    params: { catalogId: '910244132' }
  };
  defaultProps = {
    catalogItem,
    fetchCatalogIfNeeded,
    fetchApisIfNeeded,
    match
  };
  wrapper = shallow(<APIListPage {...defaultProps} />);
});

test('should render APIListPageComponent correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
