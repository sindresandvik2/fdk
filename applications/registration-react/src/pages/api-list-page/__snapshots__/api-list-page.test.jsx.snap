// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render APIListPageComponent correctly 1`] = `
<div
  className="container"
>
  <div
    className="row mb-5"
  >
    <div
      className="col-12"
    >
      <h1>
        API-katalog
      </h1>
      <div
        className="fdk-reg-datasets-publisher mt-2 mb-4"
      >
        RAMSUND OG ROGNAN REVISJON
      </div>
    </div>
  </div>
  <div
    className="row mb-5"
  >
    <div
      className="col-12"
    >
      <Link
        className="ml-0"
        replace={false}
        to="/catalogs/910244132/apis/import"
      >
        <i
          className="fa fa-plus mr-2"
        />
        Registrer ny API-beskrivelse
      </Link>
    </div>
  </div>
  <div
    className="row mb-2 mb-md-5"
  >
    <div
      className="col-12 fdk-reg-datasets-list"
    >
      <withState(withState(withHandlers(ListItems)))
        catalogId="910244132"
        defaultEmptyListText="Det er ikke registrert noen API-beskrivelser ennÃ¥."
        itemTitleField={
          Array [
            "openApi",
            "info",
            "title",
          ]
        }
        items={null}
        prefixPath="/catalogs/910244132/apis"
      />
    </div>
  </div>
</div>
`;
